R/confirmatoryfactoranalysis.R

```diff
@@ -61,29 +64,32 @@ confirmatoryFactorAnalysisInternal <- function(jaspResults, dataset, options, ..
 
 # Preprocessing functions ----
 .cfaReadData <- function(dataset, options) {
+
   if (!is.null(dataset)) return(dataset)
 
   # NOTE: The GUI does not yet allow for putting the same variable in different factors.
   # if the same variable is used twice but with a different type then this would
   # crash the R code. However, since this is not possible yet, this should be okay for now
   vars  <- unlist(lapply(options[["factors"]], `[[`, "indicators"),       use.names = FALSE)
-  types <- unlist(lapply(options[["factors"]], `[[`, "types"),            use.names = FALSE)
 
   if (length(vars) == 0)
     return(data.frame())
 
   duplicateVars <- duplicated(vars)
   vars  <- vars[!duplicateVars]
-  types <- types[!duplicateVars]
 
-  splitVars <- split(vars, types)
-  groupVar <- if (options[["group"]] == "") NULL else options[["group"]]
+  if (options[["dataType"]] == "raw") {
+    # make sure on the qml side that groupVar is indeed a nominal variable
+    groupVar <- if (options[["group"]] == "") NULL else options[["group"]]
+    dataset <- .readDataSetToEnd(columns = c(vars, groupVar))
 
-  return(.readDataSetToEnd(
-    columns.as.numeric = splitVars[["scale"]],
-    columns.as.ordinal = splitVars[["ordinal"]],
-    columns.as.factor  = groupVar
-  ))
+  } else {
+
+    dataset <- .readDataSetToEnd(all.columns = TRUE)
```

Comment:
I guess we are keeping it for this analysis then?

Reply:
see https://github.com/jasp-stats/INTERNAL-jasp/issues/2664